### ISSUE TITLE
Donate Button

### ISSUE THREAD

**Issue by defucilis** [defucilis/ScriptAxis#22]
As per mosaicmusician's suggestion, there should be a prominent donate button on each script and creator page, making it easy for people to donate to script creators who make scripts that they enjoy.

Obviously I don't want to have to worry about payment processing myself, so this would need to be done with some kind of external integration. Perhaps [Buy Me a Coffee](https://www.buymeacoffee.com/)? Once a Creator account had been verified and linked to a User, that user could add in their BMAC account credentials, and if those credentials exist, we could show the donation suggestion button on all of that Creator's script pages, as well as on their main creator page.

This would, of course, mean that there would be a financial incentive for people to abuse the account verification system to claim donations on scripts that they didn't create, so this feature is entirely dependent on that system being ironclad first. But it would be a nice stop-gap to get script creators paid before figuring out whether it's possible (and if so, how) to monetize ScriptAxis and sort out some kind of revenue distribution scheme...

**Comment by realcumber** [comment]
I think having this feature with unchecked financial incentives for sure are prone to abuse but more importantly will open up issues of directly bypassing certain conditions necessary of being able to getting paid for scripts.  
I would recommend avoiding such implementation as issues like scripts specifically created through JFS are meant to be free for personal use unless authorized at commercial script sites SexlikeReal and Realsync, where studio permissions necessary are also obtained.

**Comment by defucilis** [comment]
Commercialization and paid scripts are probably the area of the community that I'm most ignorant of, so if it's OK I have some follow-up questions (I'm very aware that features related to commercialization are the ones that can do the most damage so those are the ones where I'm trying to gather as much information as possible and move as slowly as possible on):

  * What necessary conditions are you referring to that would be bypassed? Are they something that could be built into the way that a donate feature might work?
  * Do you mean that JoyFunScripter has some terms-of-use that specifies that scripts made with it can't be sold without authorization? That's fair enough if it's the case, but I couldn't find any information anywhere on the EroScripts post, documentation or in the download folder that mentions that.
  * On what grounds do studios require permission to share and/or commercialize funscripts? The funscripts themselves don't include any copyrighted material so I can't imagine why studio permission would be required to commercialize a script, but I'm certainly no copyright expert

The way I think of a donate button isn't so much "pay me this money to download this script" so much as a "if you appreciate this work I do, feel free to send me some money as a thank-you". A donate feature wouldn't be tied in any way to the availability of scripts. For example, if a creator did something like upload an incomplete script and then put in the description "once I reach $100 in donations I'll upload the full funscript", I'd consider that a bannable abuse.

**Comment by realcumber** [comment]
"The way I think of a donate button isn't so much "pay me this money to download this script" so much as a "if you appreciate this work I do, feel free to send me some money as a thank-you". A donate feature wouldn't be tied in any way to the availability of scripts. For example, if a creator did something like upload an incomplete script and then put in the description "once I reach $100 in donations I'll upload the full funscript", I'd consider that a bannable abuse."

This is more reassuring and your further explanation helps on understanding the sites intent/overall plans and having donations seems like no issues in this manner.

I would be more concerned with the 'claiming donations' method you propose, since the site itself could be potentially indirectly profiting commercially of others work with the unclaimed donations that are received by work linked to users that may be unaware that their work is being shared.
Hence why I dont think a 'for everyone' donation button including nonverified accounts is necessarily the best approach

I've dealt with studios and scripts commercial side of it for awhile now and the consensus has been that scripts are essentially derivative works that work off the videos they are scripted for.  Donations again in the manner you mention should be ok, as long as they remain that, its just something to consider and keep in mind

**Comment by defucilis** [comment]
Yeah only 'verified creators' would be able to claim donations - the way I've set the system up at the moment is that there are two completely distinct entities - Users and Creators. Users are people with accounts, who can login, favorite scripts, etc. Users can also add scripts to the database, and when they do so they designate a Creator for that script. So each script in the database has an 'owner' (the person who created the script entry in the database) and a 'creator' the person who actually created the script.

So the verification process (once I figure out the best way to do that) is basically going to be the process by which a User is linked with a Creator. This is a 'verified user'. So only those scripts and creator pages that were associated with verified users would be able to have the donation button. And only verified users would be able to connect up a BMAC account or whatever.

And ah that's great to know re derivative works, thanks for that!
